src/TensorTwoD.hpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef TENSORTWOD_HPP_INCLUDE
+#define TENSORTWOD_HPP_INCLUDE
+
+#include "geopm/json11.hpp"
+
+#include "TensorOneD.hpp"
+
+class TensorTwoD
+{
+    public:
+        TensorTwoD();
+        TensorTwoD(int rows, int cols);
+        TensorTwoD(const TensorTwoD&);
+        TensorTwoD(json11::Json input);
+        void set_dim(int rows, int cols);
+        int get_rows() const;
+        int get_cols() const;
+        TensorOneD operator*(const TensorOneD&);
+        TensorOneD &operator[](int idx);
+        TensorOneD operator[](int idx) const;
```

Comment:
It would be a good idea to have all dimensions as `size_t` data type.
```cpp
TensorTwoD(size_t rows, size_t cols);
void set_dim(size_t rows, size_t cols);
size_t get_rows() const;
size_t get_cols() const;
TensorOneD &operator[](size_t idx);
TensorOneD operator[](size_t idx) const;
```